src/PlatformAdapter/Native/Exec/NativeExecImpl.cpp

```diff
@@ -0,0 +1,253 @@
+#include "NativeExecImpl.hpp"
+
+#include "Utils/Utils.hpp"
+
+#undef INFO
+#define INFO(msg, ...)
+
+#undef DEBUG
+#define DEBUG(msg, ...)
+
+/*
+ * NativeExecImpl
+ */
+bool NativeExecImpl::schedule_eval(const std::string& code)
+{
+  if (m_state == DEAD)
+  {
+    FAILED("#NativeExecImpl::schedule_eval, error, dead state");
+    return false;
+  }
+
+  while (m_state != RUNNING)
+  {
+  }
+
+  NativeEvalTask* task = new NativeEvalTask();
+  task->m_code = code;
+  task->m_exec_impl_ptr = this;
+
+  const std::lock_guard<std::mutex> lock(m_tasks_mutex);
+  m_tasks.insert(task);
+
+  uv_async_init(m_loop,
+                &task->m_async_task,
+                [](uv_async_t* async)
+                {
+                  uv_close(reinterpret_cast<uv_handle_t*>(async), nullptr);
+                  NativeEvalTask* task = node::ContainerOf(&NativeEvalTask::m_async_task, async);
+                  task->run();
+                });
+  uv_async_send(&task->m_async_task);
+
+  return true;
+}
+
+int NativeExecImpl::eval(std::string_view buffer)
+{
+  DEBUG("#NativeExecImpl::eval, enter");
+
+  Locker locker(m_isolate);
+  Isolate::Scope isolate_scope(m_isolate);
+  HandleScope handle_scope(m_isolate);
+
+  auto context = m_setup->context();
+  Context::Scope context_scope(context);
+
+  auto maybe_local_v8_string = v8::String::NewFromUtf8(m_isolate, buffer.data());
+  if (maybe_local_v8_string.IsEmpty())
+  {
+    FAILED("#NativeExecImpl::eval, error, script is empty");
+    return -1;
+  }
+
+  auto local_v8_string = maybe_local_v8_string.ToLocalChecked();
+  auto maybe_script = v8::Script::Compile(context, local_v8_string);
+  if (maybe_script.IsEmpty())
+  {
+    FAILED("#NativeExecImpl::eval, error, complie script error");
+    return -1;
+  }
+
+  DEBUG("#NativeExecImpl::eval, run");
+  auto script = maybe_script.ToLocalChecked();
+  auto script_result = script->Run(context);
+  if (script_result.IsEmpty())
+  {
+    FAILED("#NativeExecImpl::eval, error, run script error");
+    return -1;
+  }
+
+  DEBUG("#NativeExecImpl::eval, success");
+  return 0;
+}
+
+int NativeExecImpl::run_node(const int argc,
+                             const char** argv,
+                             std::shared_ptr<std::thread>& nodeThread,
+                             const std::function<void(const node::Environment*)>& envHook,
+                             const std::function<const char*()>& getScriptHook)
+{
+  uv_setup_args(argc, const_cast<char**>(argv));
+  std::vector<std::string> args(argv, argv + argc);
+
+  nodeThread.reset(new std::thread([&, args]() { node_main(args, envHook, getScriptHook); }));
```

Comment:
should we care about the return code of `node_main`?

Reply:
add log for debug.